fiftyone/core/dataset.py

```diff
@@ -6204,6 +6204,11 @@ def _serialize(self):
         return self._doc.to_dict(extended=True)
 
     def _update_last_loaded_at(self):
+        if self.name is None:
```

Comment:
@brimoor since this update is called every time a dataset is accessed, I think having this check is better than not. While it doesn't solve the root of the problem, it will prevent writing no-name entries to the collection for bugs we may not be aware of

Reply:
`_update_last_loaded_at` is only called in three cases:
1. in `Dataset()`
2. in `DatasetSingleton`
3. in `Dataset.reload()`

#2615 ensures that 1 and 2 will never find a `Dataset` with `name=None`.

Reply:
Only thing I think we could do is add something to `Dataset.reload()` that will mark the dataset as deleted if it reloads its data and finds that `name=None`. And *not* update `last_loaded_at`.